app/scripts/pages/file.jsx

```diff
@@ -11,14 +11,15 @@ class File extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            loading: false
+            loading: false,
+            project: ProjectStore.project
         };
     }
 
     componentDidMount() {
         let id = this.props.params.id;
         this.unsubscribe = ProjectStore.listen(state => this.setState(state));
-        ProjectActions.getFileParent(id);
+        ProjectActions.getFileContainer(id);
```

Comment:
extract into method and do a this.props.params.id check on prevProps to see if you should load the FileContainer